src/theme-preval.js

```diff
@@ -147,243 +44,33 @@ const fontSizes = ['12px', '14px', '16px', '20px', '24px', '32px', '40px', '48px
+const {scale: _excludeScaleShadows, ...shadows} = filterObject(primitives.colors.light, value => isShadowValue(value))
 
-const flash = {
-  default: {
-    backgroundColor: blue[1],
-    borderColor: 'rgba(4, 66, 137, 0.2)'
-  },
-  success: {
-    backgroundColor: green[1],
-    borderColor: 'rgba(23, 111, 44, 0.2)'
-  },
-  danger: {
-    backgroundColor: '#FFE3E6',
-    borderColor: 'rgba(158, 28, 35, 0.2)'
-  },
-  warning: {
-    backgroundColor: yellow[1],
-    borderColor: 'rgba(176, 136, 0, 0.2)'
-  }
-}
-
-// this has to be separated from the flash object since we have to use an svg selector to style the icon color
-const flashIcon = {
-  default: 'rgba(4, 66, 137, 0.6)',
-  success: 'rgba(23, 111, 44, 0.8)',
-  danger: 'rgba(158, 28, 35, 0.6)',
-  warning: yellow[8]
-}
-
-const popovers = {
-  colors: {
-    caret: 'rgba(27, 31, 35, 0.15)'
-  }
-}
-
-const pagination = {
-  borderRadius: radii[2],
-  spaceBetween: space[1],
-  colors: {
-    normal: {
-      fg: colors.gray[9]
-    },
-    disabled: {
-      fg: colors.gray[3],
-      border: 'transparent'
-    },
-    hover: {
-      border: colors.border.grayLight
-    },
-    selected: {
-      fg: colors.white,
-      bg: colors.blue[5],
-      border: 'transparent'
-    },
-    active: {
-      border: colors.border.grayLight
-    },
-    nextPrevious: {
-      fg: colors.blue[5]
-    }
-  }
-}
-
-const stateLabels = {
-  sizes: {
-    small: {
-      padding: `${space[1]} ${space[2]}`,
-      fontSize: fontSizes[0]
-    },
-    normal: {
-      padding: `${space[2]} 12px`,
-      fontSize: fontSizes[1]
-    }
-  },
-
-  status: {
-    issueClosed: {
-      backgroundColor: red[5]
-    },
-    pullClosed: {
-      backgroundColor: red[5]
-    },
-    pullMerged: {
-      backgroundColor: purple[5]
-    },
-    issueOpened: {
-      backgroundColor: '#159739' // custom green
-    },
-    pullOpened: {
-      backgroundColor: '#159739' // custom green
-    },
-    draft: {
-      backgroundColor: gray[5]
-    }
-  }
-}
-
-const {scale: _excludeScaleColors, ...functionalColors} = filterObject(primitives.colors.light, value =>
-  isColorValue(value)
-)
-const {scale: _excludeScaleShadows, ...functionalShadows} = filterObject(primitives.colors.light, value =>
-  isShadowValue(value)
-)
-
-const mergedColors = deepmerge(colors, functionalColors)
-const mergedShadows = deepmerge(shadows, functionalShadows)
+// this is a small hack that is a temporary stop gap.
+// this file must be in vanilla JS to work with preval
+// but our temporary filter utils make it impossible for
+// our TypeScript to properly infer const object structure.
+// shallow merging primitives objects back in restores the
+// structural inferences and ultimately gets overrideen with
+// the filtered values.
+const typedColors = {...primitives.colors.light, ...colors}
+const typedShadows = {...primitives.colors.light, ...shadows}
+// ---------
```

Comment:
Really simple two line fix here to fix the obscured type inferencing. This *does* add some extra cruft to the shadows object that we were previously filtering out, but if we knock out some top level keys in `primitives.colors.light` for the shadows, we would get an identical object to the original since we are doing a shallow merge. Just not sure if that was important enough to do for a temporary fix.